ui/src/criteria/attribute.tsx

```diff
@@ -0,0 +1,163 @@
+    return <AttributeEdit dispatchFn={dispatchFn} data={this.data} />;
+  }
+
+  renderDetails() {
+    return <ConceptDetails data={this.data} />;
+  };
+
+  generateFilter(entityVar: string, fromOccurrence: boolean) {
+    return null;
+  }
+
+  occurrenceEntities() {
+    return [];
+  }
+}
+
+function AttributeEdit(props: AttributeEditProps) {
+  const dispatch = useAppDispatch();
+  const underlay = useUnderlay();
+  const demographic = props.data.entities[0].name;
+  const demographicName = verifyName(demographic);
+
+  function verifyName(demographic: string) {
+    if (demographic == "sex assigned at birth") {
+      demographic = "sex_at_birth";
+    } else if (demographic == "gender identity") {
+      demographic = "gender";
+    }
+    return demographic.concat("_concept_id");
+  }
+
+  const categoryList = underlay?.entities
+    .filter((g) => g.name === "person")[0]
+    .attributes?.filter((g) => g.name === demographicName)[0]
+    .attributeFilterHint?.enumHint?.enumHintValues;
+  const index =
+    categoryList?.map((hint) => {
+      return props.data.selected.findIndex(
+        (row) =>
+          row.entity === demographic &&
+          row.id === (hint.attributeValue?.int64Val as number)
+      );
+    }) || [];
+  console.log(index);
+
+  return (
+    <>
+      {categoryList?.map((hint, idx) => (
+        <ListItem key={hint.displayName || ""}>
+          <FormControlLabel
+            label={hint.displayName || ""}
+            control={
+              <Checkbox
+                size="small"
+                checked={index[idx] > -1}
+                onChange={() => {
+                  props.dispatchFn(
+                    produce(props.data, (data) => {
+                      console.log(idx, index[idx]);
+                      if (index[idx] > -1) {
+                        data.selected.splice(index[idx], 1);
+                      } else {
+                        const name = hint.displayName;
+                        data.selected.push({
+                          entity: demographic,
+                          id: hint.attributeValue?.int64Val as number,
+                          name: !!name ? String(name) : "",
+                        });
+                      }
+                    })
+                  );
+                }}
+              />
+            }
+          />
+        </ListItem>
+      ))}
+    </>
+  );
+}
+
+type ConceptDetailsProps = {
+  data: Data;
+};
+
+function ConceptDetails(props: ConceptDetailsProps) {
+  return (
+    <>
+      {props.data.selected.length === 0 ? (
+        <Typography variant="body1">None selected</Typography>
+      ) : (
+        props.data.selected.map(({ id, name }) => (
+          <Stack direction="row" alignItems="baseline" key={id}>
+            <Typography variant="body1">{id}</Typography>&nbsp;
+            <Typography variant="body2">{name}</Typography>
+          </Stack>
+        ))
+      )}
+    </>
+  );
+}
```

Comment:
The usual standard to end files with newlines. See if you can configure your editor to do it automatically.

Reply:
Got it. Done from setting-general-every save with one line